pkg/operator/api/api.go

```diff
@@ -74,6 +99,43 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+func (c *Cluster) makeClusterRole() error {
+	account := &v1.ServiceAccount{}
+	account.Name = DeploymentName
+	account.Namespace = c.Namespace
+	_, err := c.context.Clientset.CoreV1().ServiceAccounts(c.Namespace).Create(account)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create api service account. %+v", err)
+	}
+
+	// Create the cluster role if it doesn't yet exist.
+	// If the role already exists we have to update it. Otherwise if the permissions change during an upgrade,
+	// the create will fail with an error that we're changing the permissions.
+	role := &v1beta1.ClusterRole{Rules: clusterAccessRules}
+	role.Name = DeploymentName
+	_, err = c.context.Clientset.RbacV1beta1().ClusterRoles().Get(role.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
if the kubernetes object is not found, an error is returned, therefore errors.IsNotFound(err) is not the same as err == nil.